staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -456,3 +456,15 @@ func applyPatchToObject(
 
 	return nil
 }
+
+// interpretPatchError interprets the error type and returns an error with appropriate HTTP code.
+func interpretPatchError(err error) error {
+	switch err {
```

Comment:
`switch err.(type) {`, no?

Reply:
https://tour.golang.org/methods/16

Reply:
`mergepatch.ErrBadJSONDoc`, etc are all of type `error`. So `err.(type)` will give `error` everytime.

Since `mergepatch.ErrBadJSONDoc` is already of type `error`, we need to check the values (instead of a type switch)

Reply:
oh, so these are all constant error objects? Then your code is right. :+1:

Reply:
> oh, so these are all constant error objects? 

Yes, they are constant error objects formatted with the given text. :)